### PR TITLE
[NCL-8234] - Fix the math of the DalAnalyzer report; compute the PURL…

### PR DESCRIPTION
**Pull request by vibe13**
… for the generic not found artifacts

### Checklist:

* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java [hunk 1]
```diff
@@ -259,6 +259,7 @@ private org.jboss.pnc.model.Artifact findOrCreateArtifact(
 
         // create
         artifact.setTargetRepository(targetRepo);
+        createGenericPurl(targetRepo.getRepositoryPath(), artifact.getFilename().toString(), artifact.getSha256());
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java:262]: Do you just throw away the generated purl?

Comment (vibe13) [reply facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java:262]: From my understanding of the code, at this point there is no generated PURL for the not found artifacts yet, so I am not throwing anything away?

File: facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java [hunk 2]
```diff
@@ -364,6 +366,29 @@ private String createPURL(MavenArtifact mavenArtifact) {
...
 
+    /**
+     * Compute the purl string for a generic download, that does not match package type specific files structure. See
+     * https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#generic
+     *
+     * @param downloadUrl url where the artifact was downloaded from
+     * @param filename the artifact filename
+     * @param sha256 the SHA-256 of the artifact
+     * @return the generated purl
+     * @throws MalformedPackageURLException
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java:377]: it doesn't throw that

File: facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java (hunk 1 above)
Comment (janinko) [reply facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java:262]: The method returns String and has no sideffect (except for the posible exception) and the return value is not used. So this line basically does nothing?

Comment (vibe13) [reply facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java]: doh of course

File: facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java (hunk 2 above)
Comment (vibe13) [reply facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java]: good catch

**Review by janinko — APPROVED** [review]

**Comment by vibe13** [comment]
retest this please

**Pull request merged**